sys/riscv/riscv/pmap.c

```diff
@@ -3536,6 +3542,8 @@ pmap_caploadgen_next(pmap_t pmap)
  * aren't. (Toooba in particular doesn't yet do that.)  If we ever get here,
  * update the bits of pmap_caploadgen_update and cheri_pte_cr that refers to
  * this note, too.
+ *
+ * XXXMJ as with pmap_sync_capdirty(), this only traverses managed mappings.
```

Comment:
*That* is much more of a problem, since we really do need to know that all aliases are not capability bearing in order for this dance to be safe.  Is there a flag for "has an unmanaged mapping" we could use to back off to leave such pages always under the revoker's watch?  That's not a _stellar_ answer, but it'd be a correct baseline.  Open to other suggestions as well!

Reply:
This is actually a property of the physical page. That is, a page itself is either managed or not (`m->oflags & VPO_UNMANAGED`), and in general all mappings of a managed page will be managed. It is however possible in principle for the kernel to create unmanaged mappings of any page (usually through pmap_qenter()). I'm not aware of anything that does this with pages mapped into a userspace process, but I believe there is nothing preventing it. I tried adding an assertion to pmap_qenter() at least and nothing blew up. Perhaps we should enforce that.